src/Atmos/Model/source.jl

```diff
@@ -65,11 +65,7 @@ function atmos_source!(
     t::Real,
     direction,
 )
-    if atmos.ref_state isa HydrostaticState
-        source.ρu -= (state.ρ - aux.ref_state.ρ) * aux.orientation.∇Φ
-    else
-        source.ρu -= state.ρ * aux.orientation.∇Φ
-    end
+    source.ρu -= state.ρ * aux.orientation.∇Φ
```

Comment:
Same as above: Why is the subtraction of hydrostatic balance removed from the momentum equation? Or does it occur somewhere else now where I missed it?

Reply:
No, the only cancel out in your initial state, where ρ and p are hydrostatically balanced. But this is a special initial state, and not one that is generally maintained. In general, there is a nonhydrostatic residual, and this is essential (e.g., it's what's responsible for convection).

Reply:
Same as above: I reverted this change